### ISSUE TITLE
Configuration of DNS Over TLS communication protocols and cipher suites

### ISSUE THREAD

**Issue by techpavan** [systemd/systemd#30097]
### Component

systemd-networkd

### Is your feature request related to a problem? Please describe

Currently there does not seem to be an option to choose the exact TLS version that can be used for DNS Over TLS resolutions. Even though TLS 1.3 is supported, there is no way to restrict the communication from falling back to TLS 1.2. Also, no way to control the cipher suites used in this communication.

Adding these controls for configuration might help in a secure environment where a granular control is required.

### Describe the solution you'd like

Be able to configure TLS protocols and cipher suites.

**Comment by topimiettinen** [comment]
I'd make it easy to be secure, so that only the latest TLS is allowed by default and the administrator has to explicitly configure weaker versions with something like `AllowTLSVersions=1.2`.

**Comment by techpavan** [comment]
True to a larger extent. But there are cases like FIPS where latest TLS 1.3 does not still comply with FIPS. There may be organizations that prefer to stick with N-1 until N is widely confirmed to be stable.

So, default can be latest, but the moment `AllowTLSVersions` exist in configuration that should imply, "only" use what is configured - nothing more, nothing less.

**Comment by topimiettinen** [comment]
Yes, perhaps the name should then be `AllowTLSVersionsOnly=1.2` or `TLSVersionAllowList=1.2` to reflect that.

**Comment by topimiettinen** [comment]
In the API for OpenSSL, the restrictions can be set as a minimum and maximum TLS levels. We already set the minimum to 1.2 [here](https://github.com/systemd/systemd/blob/b6d669d5af5b23550c7592432bcfed2ff73c362b/src/resolve/resolved-dnstls-openssl.c#L402). It's probably easiest to reflect the API with something like `TLSMinimumProtocolVersion=1.2` and `TLSMaximumProtocolVersion=1.2`.
